src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,32 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        if (robot.getX() != toX) {
+            setHorizontalDirection(robot, toX);
+            while (robot.getX() != toX) {
```

Comment:
we are checking the same condition twice

Reply:
got it